server/core/templates/cred_update/add_password_modal_partial.html

```diff
@@ -6,37 +6,70 @@ <h5 class="modal-title" id="staticPasswordLabel">Add a New Password</h5>
                 <button type="button" class="btn-close" aria-label="Close"></button>
             </div>
             <div class="modal-body">
-                <form class="row g-3 needs-validation" novalidate
-                      hx-post="/ui/reset/pw_check"
-                      hx-trigger="input from:#new-password delay:200ms changed, input from:#new-password-check delay:200ms changed">
-                    <input hidden=true type="text" autocomplete="username" />
+                <form class="row g-3 needs-validation" id="newPasswordForm" novalidate
+                      hx-on::after-request="submitResponseHandler(event)"
+                      hx-on::before-swap="stillSwapFailureResponse(event)">
+                    <input hidden type="text" autocomplete="username" />
+                    (% let potentially_invalid_input_class = "" %)
+                    (% let potentially_invalid_reinput_class = "" %)
+                    (% let pwd_equal = true %)
+
+                    (% if let PwdCheckResult::Failure with { pwd_equal, warnings } = check_res %)
+                        (% let pwd_equal = pwd_equal.clone() %)
+                        (% let potentially_invalid_input_class = "is-invalid"|blank_if(warnings.len() == 0) %)
+                        (% let potentially_invalid_reinput_class = "is-invalid"|blank_iff(pwd_equal) %)
+                    (% endif %)
+
                     <label for="new-password" class="form-label">Enter New Password</label>
                     <input
                             aria-describedby="password-validation-feedback"
                             autocomplete="new-password"
-                            class="form-control"
+                            class="form-control ((potentially_invalid_input_class))"
+                            name="new_password"
                             id="new-password"
                             placeholder=""
                             type="password"
                             required
+                            autofocus
                     />
+                    <!-- bootstrap hides the feedback if we remove is-invalid from the input above -->
+                    <div id="password-validation-feedback" class="invalid-feedback d-block">
+                        <ul>
+                            (% if let PwdCheckResult::Failure with { pwd_equal, warnings } = check_res %)
```

Comment:
Move this block outside the `<div>` above it, don't need the tags if we don't have the `check_res` Failure to handle 😄